scene/main/scene_tree.cpp

```diff
@@ -576,6 +577,12 @@ bool SceneTree::idle(float p_time) {
 
 	flush_transform_notifications(); //additional transforms after timers update
 
+	// flush pending camera scrolls
+	Camera2D::flush_pending_scrolls();
```

Comment:
I'm not fond of how this breaks encapsulation by having the SceneTree responsible for flushing a particular node. Might be OK as a temporary workaround but I'm not sure it's a good design for the long term.

Reply:
This could be moved to Viewport. Somewhat related: #38317

Reply:
I wasn't absolutely sure on this either. It just needs a good place to flush it, so welcome ideas. Reduz also suggested looking at setting priority for process, but I was a bit worried about order of operations. Something in the process might have needed an update transforms before the camera updates, and we'd be back to the problem of a frame delay.

So this pattern of update everything else -> flush transforms -> scroll cameras -> flush transforms could be crucial for it to work perfectly.